src/inspector_socket_server.h

```diff
@@ -30,17 +30,27 @@ class SocketServerDelegate {
   virtual void ServerDone() = 0;
 };
 
+// HTTP Server, writes messages requested as TransportActions, and responds
+// to HTTP requests and WS upgrades.
 class InspectorSocketServer {
  public:
   using ServerCallback = void (*)(InspectorSocketServer*);
   InspectorSocketServer(SocketServerDelegate* delegate,
                         const std::string& host,
                         int port,
                         FILE* out = stderr);
+  // XXX(sam) why is loop not passed in ctor? Its fixed and known at that time.
```

Comment:
Am I missing something? It seems like most of the state is passed in the ctor, and then some is passed in Start(), and I don't see why, is it just historical? Is there some future use-case this is designed around?